pkg/codegen/go/gen_program_expressions.go

```diff
@@ -814,9 +858,14 @@ func (g *generator) genApply(w io.Writer, expr *model.FunctionCallExpression) {
 	isInput := false
 	retType := g.argumentTypeName(nil, then.Signature.ReturnType, isInput)
 	// TODO account for outputs in other namespaces like aws
-	typeAssertion := fmt.Sprintf(".(%sOutput)", retType)
-	if !strings.HasPrefix(retType, "pulumi.") {
-		typeAssertion = fmt.Sprintf(".(pulumi.%sOutput)", Title(retType))
+	var typeAssertion string
+	if retType == "[]string" {
```

Comment:
Calling this out, something is really patchy/incomplete here even after the fix in the PR. Does it ring any bells?